campaigntools.php

```diff
@@ -27,35 +27,32 @@ function campaigntools_civicrm_buildForm($formName, &$form) {
       || $formName === 'CRM_Event_Form_Registration_Register'
       || $formName === 'CRM_Event_Form_Registration_Confirm'
     ) {
-    // List all campaign ids available
-    $campaign = array(
-      'ids' => array(
-        1, 2,
-      ),
-    );
+
     // Get the campaign id by the entryURL in the controler array
     $controller = $form->getVar('controller');
     // Process to get the campaign id in the parameter
     $params = explode('?', $controller->_entryURL);
     parse_str(end($params), $parseURL);
-    $paramItems = [];
 
     // Remove amp; since it was not remove using parse_str
     foreach($parseURL as $key => $value) {
       $newKey = str_replace('amp;', '', $key);
-      $paramItems[$newKey] = $value;
-    }
-
-    // Check if campaign id is set and match it on the ones in campaign_ids
-    if (isset($paramItems['campaign']) && in_array($paramItems['campaign'], $campaign['ids'])) {
-      // $form->setVar is not working so I used this one
-      // set campaign_id in values array so it will be save in the database
 
-      // If its an event, save it under values event array
-      if(!empty($form->_values['event'])) {
-        $form->_values['event']['campaign_id'] = $paramItems['campaign'];
-      } else {
-        $form->_values['campaign_id'] = $paramItems['campaign'];
+      // Check if campaign param exist
+      if ($newKey === 'campaign') {
+        // Check campaign param if exist on database
+        $campaign = civicrm_api3('Campaign', 'getcount', [
```

Comment:
Minor improvement in code quality for other developers in the future:
Please name this variable `$campaignCount` -- which indicates clearly that it's an integer or a count of something, instead of `$campaign` -- which suggests that it might be an array representing various attributes of a campaign record.

Reply:
Done